sdks/java/io/kafka/src/test/java/org/apache/beam/sdk/io/kafka/KafkaIOTest.java

```diff
@@ -410,6 +441,68 @@ public void testUnboundedSourceCheckpointMark() throws Exception {
   }
 
   @Test
+  public void testUnboundedSourceCheckpointMarkWithEmptyPartitions() throws Exception {
+    // Similar to testUnboundedSourceCheckpointMark(), but verifies that source resumes
+    // properly from empty partitions, without missing messages added since checkpoint.
+
+    // Initialize consumer with than number of partitions so that some of them are empty.
```

Comment:
fixed. I wanted to say 'Initialize consumer with fewer elements than number of partitions...'